### PR TITLE
ECL-405 Unincorporated Association Entity Type

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import uk.gov.hmrc.economiccrimelevyregistration.models.Mode
+import uk.gov.hmrc.economiccrimelevyregistration.navigation.AddCTUTRPageNavigator
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.AddCTUTRView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AddCTUTRController @Inject() (
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala:34]:
```suggestion
class CtUtrController @Inject() (
```

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/AddCTUTRFormProvider.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.forms
+
+import play.api.data.Form
+import uk.gov.hmrc.economiccrimelevyregistration.forms.mappings.Mappings
+import uk.gov.hmrc.economiccrimelevyregistration.forms.mappings.MaxLengths.CTUTRLength
+
+class AddCTUTRFormProvider extends Mappings {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/AddCTUTRFormProvider.scala:23]:
```suggestion
class CtUtrFormProvider extends Mappings {
```

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/AddCTUTRPageNavigator.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.navigation
+import play.api.mvc.Call
+import uk.gov.hmrc.economiccrimelevyregistration.models.Registration
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.routes
+
+class AddCTUTRPageNavigator extends PageNavigator {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/navigation/AddCTUTRPageNavigator.scala:22]:
```suggestion
class CtUtrPageNavigator extends PageNavigator {
```

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CTUTRPostcodePageNavigator.scala [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.navigation
+
+import play.api.mvc.Call
+import uk.gov.hmrc.economiccrimelevyregistration.models.Registration
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.routes
+
+class CTUTRPostcodePageNavigator extends PageNavigator {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CTUTRPostcodePageNavigator.scala:23]: PascalCase / UpperCamelCase comment applies

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/MaxLengths.scala [hunk 5]
```diff
@@ -24,5 +24,6 @@ object MaxLengths {
   val RoleMaxLength                   = 160
   val OrganisationNameMaxLength       = 160
   val CompanyRegistrationNumberLength = 8
-
+  val CTUTRLength                     = 10
+  val CTUTRPostcodeLength             = 8
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/MaxLengths.scala:28]: PascalCase / UpperCamelCase comment applies

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/CTUTRPostcodeFormProvider.scala [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.forms
+
+import play.api.data.Form
+import uk.gov.hmrc.economiccrimelevyregistration.forms.mappings.Mappings
+import uk.gov.hmrc.economiccrimelevyregistration.forms.mappings.MaxLengths.CTUTRPostcodeLength
+class CTUTRPostcodeFormProvider extends Mappings {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/CTUTRPostcodeFormProvider.scala:22]: PascalCase / UpperCamelCase comment applies

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import uk.gov.hmrc.economiccrimelevyregistration.models.Mode
+import uk.gov.hmrc.economiccrimelevyregistration.navigation.DoYouHaveCTUTRPageNavigator
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.DoYouHaveCTUTRView
+
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DoYouHaveCTUTRController @Inject() (
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:35]: PascalCase / UpperCamelCase comment applies

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala [hunk 8]
```diff
@@ -273,4 +276,23 @@ trait EclTestData {
...
+                                   None,
+                                   None,
+                                   Some("0123456789"),
+                                   None,
+                                   None,
+                                   Some("A1B YT4")
+                                 )
+      } yield RegistrationWithUnincorporatedAssociation(registration =
+        registration.copy(optOtherEntityJourneyData = Some(otherEntityJourneyData))
+      )
```
Comment (peter-hazell) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala:296]: should you also set entity type to 'Other' ?

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala [hunk 9]
```diff
@@ -99,6 +99,12 @@ trait Generators {
       chars  <- listOfN(length, Gen.alphaNumChar)
     } yield chars.mkString
 
+  def numStringsWithConcreteLength(length: Int): Gen[String] =
+    for {
+      length <- choose(length, length)
+      chars  <- listOfN(length, Gen.numChar)
+    } yield chars.mkString
```
Comment (peter-hazell) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/generators/Generators.scala:106]: I think Patrick already defined a method like this

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val ctutr: Option[String]  = if (answer) Some(answer.toString) else None
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:63]: Need a new `Option[Boolean]` on `OtherEntityJourneyData` for this.

File: conf/application.conf [hunk 11]
```diff
@@ -177,7 +177,7 @@ features {
       "654321"
     ]
     contactEmail = "economiccrimelevyenquiries@hmrc.gov.uk"
-    enabled = true
+    enabled = false
```
Comment (paddynski-opencast) [inline conf/application.conf:180]: Please undo this.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala (hunk 10 above)
Comment (peter-hazell) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:63]: Agree with Patrick - all user answers need to be represented in the journey data with the appropriate type

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CTUTRPostcodeController.scala [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  pageNavigator: CTUTRPostcodePageNavigator,
+  view: CTUTRPostcodeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[String]               = formProvider()
+  def onPageLoad(): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData) { implicit request =>
+      Ok(view(form))
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CTUTRPostcodeController.scala:50]: Please populate the form here.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+)(implicit
+  ec: ExecutionContext
+) extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData) { implicit request =>
+      Ok(view(form, mode))
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:53]: Please populate the form here.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val ctutr: Option[String]  = if (answer) Some(answer.toString) else None
+            val otherEntityJourneyData =
+              request.registration.otherEntityJourneyData.copy(ctUtr = ctutr, postcode = None)
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:65]: That is mainly because there is a case when user can come back here from _CheckYourAnswers_ page and if user clicks _Yes_ he will be prompted to add CT UTR again. If user says _No_ we are erasing previous CTUTR so it is not shown on CheckYourAnswers page.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala (hunk 10 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCTUTRController.scala:63]: I've considered same thing while I developing this but went in another direction. My train of thoughts was: If user has CT UTR it will be Some("CTUTR") where _Some_ indicates affirmative answer given and _"CTUTR"_ value provided by user. I will commit the change that you have advised.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/AddCTUTRPageNavigator.scala (hunk 3 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/AddCTUTRPageNavigator.scala]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CTUTRPostcodePageNavigator.scala (hunk 4 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CTUTRPostcodePageNavigator.scala]: Fixed.

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala (hunk 8 above)
Comment (nemanjatomanovic) [reply test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CTUTRPostcodeController.scala [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
+          postcode => {
+            val otherEntity: OtherEntityJourneyData =
+              request.registration.optOtherEntityJourneyData.get.copy(postcode = Some(postcode))
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CTUTRPostcodeController.scala]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          ctutr => {
+            val otherEntity = request.registration.optOtherEntityJourneyData.get.copy(ctUtr = Some(ctutr));
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala]: Fixed.

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala [hunk 17]
```diff
@@ -273,4 +276,23 @@ trait EclTestData {
...
+        otherEntityJourneyData = OtherEntityJourneyData(
+                                   Some(UnincorporatedAssociation),
+                                   Some("UA Test"),
+                                   None,
+                                   None,
+                                   None,
+                                   Some("0123456789"),
+                                   None,
+                                   None,
+                                   Some("A1B YT4")
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+  view: AddCTUTRView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[String] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData) { implicit request =>
+      Ok(view(form, mode))
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AddCTUTRController.scala]: Fixed.

File: conf/application.conf (hunk 11 above)
Comment (nemanjatomanovic) [reply conf/application.conf]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/EntityTypeView.scala.html [hunk 19]
```diff
@@ -16,6 +16,7 @@
 
 @import uk.gov.hmrc.economiccrimelevyregistration.viewmodels.LegendSize
 @import uk.gov.hmrc.economiccrimelevyregistration.config.AppConfig
+@import uk.gov.hmrc.govukfrontend.views.html.components.GovukRadios
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/views/EntityTypeView.scala.html:19]: Why this line? It works without it.

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala [hunk 20]
```diff
@@ -273,4 +276,27 @@ trait EclTestData {
...
   )
 
+  implicit val arbRegistrationWithUnincorporatedAssociation: Arbitrary[RegistrationWithUnincorporatedAssociation] =
+    Arbitrary {
+      for {
+        registration          <- Arbitrary.arbitrary[Registration]
+        businessName          <- Arbitrary.arbitrary[String]
+        ctutr                 <- Arbitrary.arbitrary[String]
+        postcode              <- Arbitrary.arbitrary[String]
+        otherEntityJourneyData = OtherEntityJourneyData(
```
Comment (paddynski-opencast) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala:286]: Did you forget the boolean you added to `OtherEntityJourneyData`?

File: test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala [hunk 21]
```diff
@@ -273,4 +276,27 @@ trait EclTestData {
...
+                                   Some(UnincorporatedAssociation),
+                                   Some(businessName),
+                                   None,
+                                   None,
+                                   None,
+                                   Some(ctutr),
+                                   Some(true),
```
Comment (paddynski-opencast) [inline test-common/uk/gov/hmrc/economiccrimelevyregistration/EclTestData.scala:293]: This needs to be generated.

File: test/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigatorSpec.scala [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+          mockAddressLookupFrontendConnector
+            .initJourney(ArgumentMatchers.eq(contactAddressIsUk), ArgumentMatchers.eq(mode))(any())
+        )
+          .thenReturn(Future.successful(journeyUrl))
+
+        pageNavigator.nextPage(mode, updatedRegistration) shouldBe Call(GET, journeyUrl)
+    }
+  }
+
+}
```
Comment (paddynski-opencast) [inline test/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigatorSpec.scala:50]: Remove anything to do with contact address as you are testing CTUTR page navifator.

File: test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala [hunk 23]
```diff
@@ -50,7 +50,7 @@ class EmailServiceSpec extends SpecBase {
           ViewUtils.formatLocalDate(EclTaxYear.dueDate, translate = false)(messages),
           "true",
           None,
-          appConfig.privateBetaEnabled.toString
+          (!appConfig.returnsJourneyEnabled).toString
```
Comment (paddynski-opencast) [inline test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala:53]: Why did you change this?

File: test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala [hunk 24]
```diff
@@ -98,7 +98,7 @@ class EmailServiceSpec extends SpecBase {
           eclDueDate,
           "true",
           Some(secondContactEmail),
-          appConfig.privateBetaEnabled.toString
+          (!appConfig.returnsJourneyEnabled).toString
```
Comment (paddynski-opencast) [inline test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala:101]: Why did you change this?

File: test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala [hunk 25]
```diff
@@ -108,7 +108,7 @@ class EmailServiceSpec extends SpecBase {
           eclDueDate,
           "false",
           Some(secondContactEmail),
-          appConfig.privateBetaEnabled.toString
+          (!appConfig.returnsJourneyEnabled).toString
```
Comment (paddynski-opencast) [inline test/uk/gov/hmrc/economiccrimelevyregistration/services/EmailServiceSpec.scala:111]: Why did you change this?

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package uk.gov.hmrc.economiccrimelevyregistration.navigation
+import play.api.mvc.Call
+import uk.gov.hmrc.economiccrimelevyregistration.models.Registration
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.routes
+
+class CtUtrPageNavigator extends PageNavigator {
+  override protected def navigateInNormalMode(registration: Registration): Call =
+    routes.CtUtrPostcodeController.onPageLoad()
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala]: Done.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+
+class DoYouHaveCtUtrPageNavigator @Inject() (implicit
+  ex: ExecutionContext
+) extends PageNavigator {
+  override protected def navigateInNormalMode(registration: Registration): Call = navigateInEitherMode(registration)
+
+  override protected def navigateInCheckMode(registration: Registration): Call = navigateInEitherMode(registration)
+
+  private def navigateInEitherMode(registration: Registration): Call =
+    if (registration.otherEntityJourneyData.isCtUtrPresent.get) {
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala]: Fixed.

File: app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala [hunk 28]
```diff
@@ -0,0 +1,74 @@
...
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val otherEntityJourneyData =
+              request.registration.otherEntityJourneyData.copy(isCtUtrPresent = Some(answer), postcode = None)
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala:65]: Dy you need to reset the postcode here?

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala [hunk 29]
```diff
@@ -94,14 +94,13 @@ trait Constraints {
         Valid
     }
 
-  protected def nonEmptySet(errorKey: String): Constraint[Set[_]] =
+  protected def nonEmptySet(errorKey: String): Constraint[Set[_]]                                       =
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala:97]: Not sure what you did here. Seems that you added white space only. Please remove if this is so.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala [hunk 30]
```diff
@@ -0,0 +1,47 @@
...
+) extends PageNavigator {
+  override protected def navigateInNormalMode(registration: Registration): Call =
+    registration.otherEntityJourneyData.isCtUtrPresent match {
+      case Some(isCtUtrPresent) => navigateInEitherMode(isCtUtrPresent, NormalMode)
+      case None                 => routes.NotableErrorController.answersAreInvalid()
+    }
+
+  override protected def navigateInCheckMode(registration: Registration): Call =
+    registration.otherEntityJourneyData.isCtUtrPresent match {
+      case Some(isCtUtrPresent) => navigateInEitherMode(isCtUtrPresent, CheckMode)
```
Comment (paddynski-opencast) [inline app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala:37]: If in check mode, you always go to check your answers.

File: app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala (hunk 29 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/forms/mappings/Constraints.scala:97]: That was done by scalafmtAll function.

File: app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala (hunk 30 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyregistration/navigation/DoYouHaveCtUtrPageNavigator.scala:37]: In this case it's a bit different. If user says "Yes" it should go to `CtUtrController`. If we went straight to check your answers we would never offer him a chance to enter the CT UTR if he said "Yes". Would you agree with that ?
